portfolio/src/main/webapp/scripts/comments.js

```diff
@@ -79,32 +113,39 @@ function getComments() {
         addComment(comment)
       );
     });
-    Object.keys(data).length > 0 ? toggleDeleteAllButton(true) : toggleDeleteAllButton(false);
+    // show "delete all" button if comments are present
+    const deleteAllButton = document.getElementById("delete-all-button");
+    deleteAllButton.style.display = Object.keys(data).length > 0 ? "block" : "none";
   })
 }
 
-function submitComment() {
+/* Adds a comment to datastore and prompt a reload of comments */
+async function submitComment() {
+  // get current user
+  const email = await isLoggedIn();
+  if (email === undefined) return;
+
+  // get message text
+  const message = document.getElementById("comment-input-message").value;
+
   // send ajax request to store comment
   var http = new XMLHttpRequest();
   http.open("POST", "/add-comment", true);
   http.setRequestHeader("Content-type","application/x-www-form-urlencoded");
+  http.send("email=" + email + "&message=" + message);
```

Comment:
I don't think you should have email be part of the HTTP request. This isn't really authentication because anyone could theoretically still spoof the request.

I think the intended approach is for the ```/add-comment``` servlet itself use the UserService to fetch the current user's email.